modules/Bio/EnsEMBL/Hive/HivePipeline.pm

```diff
@@ -544,151 +557,197 @@ sub print_diagram {
+                    $tweakStructure->{Return}->{NewValue} = undef;
+                    push @response, "Tweak.Deleting\tanalysis[$analysis_name].param[$param_name] ::\t".stringify($param_hash->{ $param_name })." --> (missing value)\n";
 
                     delete $param_hash->{ $param_name };
                     $analysis->parameters( stringify($param_hash) );
                     $need_write = 1;
                 } else {
+                    $tweakStructure->{Return}->{NewValue} = $new_value_str;
                     if(exists($param_hash->{ $param_name })) {
-                        print "Tweak.Changing\tanalysis[$analysis_name].param[$param_name] ::\t".stringify($param_hash->{ $param_name })." --> $new_value_str\n";
+                        push @response, "Tweak.Changing\tanalysis[$analysis_name].param[$param_name] ::\t".stringify($param_hash->{ $param_name })." --> $new_value_str\n";
                     } else {
-                        print "Tweak.Adding  \tanalysis[$analysis_name].param[$param_name] ::\t(missing value) --> $new_value_str\n";
+                        push @response, "Tweak.Adding  \tanalysis[$analysis_name].param[$param_name] ::\t(missing value) --> $new_value_str\n";
                     }
 
                     $param_hash->{ $param_name } = $new_value;
                     $analysis->parameters( stringify($param_hash) );
                     $need_write = 1;
                 }
+                push @{$responseStructure->{Tweaks}}, $tweakStructure;
             }
 
+
         } elsif($tweak=~/^analysis\[([^\]]+)\]\.(wait_for|flow_into)(\?|#|\+?=(.+))$/) {
+
             my ($analyses_pattern, $attrib_name, $operation, $new_value_str) = ($1, $2, $3, $4);
             $operation=~/^(\?|#|\+?=)/;
             my $operator = $1;
 
             my $analyses = $self->collection_of( 'Analysis' )->find_all_by_pattern( $analyses_pattern );
-            print "Tweak.Found   \t".scalar(@$analyses)." analyses matching the pattern '$analyses_pattern'\n";
+            push @response, "Tweak.Found   \t".scalar(@$analyses)." analyses matching the pattern '$analyses_pattern'\n";
 
             my $new_value = destringify( $new_value_str );
 
             foreach my $analysis (@$analyses) {
-
+                my $tweakStructure;
+                $tweakStructure->{Object}->{Type} = "Analysis";
                 my $analysis_name = $analysis->logic_name;
-
+                $tweakStructure->{Object}->{Id} = $analysis->dbID + 0;
+                $tweakStructure->{Object}->{Name} = $analysis->logic_name;
                 if( $attrib_name eq 'wait_for' ) {
-
+                    $tweakStructure->{Return}->{Field} = 'wait_for';
                     my $cr_collection   = $self->collection_of( 'AnalysisCtrlRule' );
                     my $acr_collection  = $analysis->control_rules_collection;
-
+                    $tweakStructure->{Return}->{OldValue} = join(', ', map { $_->condition_analysis_url } @$acr_collection );
```

Comment:
Since `Bio::EnsEMBL::Hive::Utils::PCL::parse_wait_for` (mostly) expects an array-ref, `OldValue` should be shown in the same format:
```perl
$tweakStructure->{Return}->{OldValue} = [map { $_->condition_analysis_url } @$acr_collection]; 
```